packages/edit-site/src/components/routes/redirect-to-homepage.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import history from '../../utils/history';
+import getIsListPage from '../../utils/get-is-list-page';
+
+function getNeedsHomepageRedirect( params ) {
+	const { postType } = params;
+	return (
+		! getIsListPage( params ) &&
+		! [ 'post', 'page', 'wp_template', 'wp_template_part' ].includes(
+			postType
+		)
+	);
+}
+
+async function getHomepageParams( siteUrl ) {
+	const siteSettings = await apiFetch( { path: '/wp/v2/settings' } );
+	if ( ! siteSettings ) {
+		return;
+	}
+
+	const {
+		show_on_front: showOnFront,
+		page_on_front: frontpageId,
+	} = siteSettings;
+
+	// If the user has set a page as the homepage, use those details.
+	if ( showOnFront === 'page' ) {
+		return {
+			postType: 'page',
+			postId: frontpageId,
+		};
+	}
```

Comment:
Most of this code is replicating `showHomepage`, but I wasn't sure how to test this as there's no customizer where this setting can be changed. Is it still possible to configure a homepage in this way?

Reply:
I think you can change some of these via Settings -> Reading -> Your homepage displays.